sat/cli/bootsys/platform.py

```diff
@@ -225,6 +227,30 @@ def has_completed(self):
         return current_state == self.target_state
 
 
+class KubernetesAPIWaiter(Waiter):
+    """Waiter subclass for waiting until Kubernetes API is reachable."""
+
+    def __init__(self, timeout, poll_interval=5):
+        super().__init__(timeout, poll_interval=poll_interval)
+
+    def has_completed(self):
+        """Check if the Kubernetes API is reachable."""
+        try:
+            # Try to list nodes using Kubernetes API
+            core_api = CoreV1Api()
```

Comment:
Use the utility function already used elsewhere in this file to get a properly initialized Kubernetes client object.

```suggestion
            core_api = load_kube_api()
```